src/compiler/Stage/Desugar.elm

```diff
@@ -195,9 +156,21 @@ desugarTuple3 recurse e1 e2 e3 =
     Canonical.Lambda arg1 (Canonical.Lambda arg2 body)
 
 -}
-curryLambda : List VarName -> Canonical.Expr -> Canonical.Expr
+curryLambda : List VarName -> Canonical.LocatedExpr -> Canonical.LocatedExpr
```

Comment:
I believe this still has the problem of the child lambdas having the Location of the body, not the parent lambda.
```elm
\x y z -> x + y + z
-->
\x -> \y -> \z -> x + y + z
```
We want all three lambdas to have the same position: the one of the original big lambda, not of the body.

I believe this can be done by passing the original position as an argument to this `curryLambda`. The big `desugarExpr` function that calls this has access to that original position.